lib/orm.php

```diff
@@ -2196,6 +2196,67 @@ public function save() {
 		return $success;
 	}
 
+	/**
+	 * Inserts multiple rows in a single query or updates single rows separately if they are not new entries.
+	 *
+	 * @param array $models Array of model instances to be inserted/updated.
+	 *
+	 * @return int|false The number of inserted rows on success, false for failure.
+	 *
+	 * @example From the Indexable_Link_Builder class: $this->seo_links_repository->query()->save_many( $links );
+	 *
+	 * @throws \Exception Primary key ID contains null value(s).
+	 * @throws \Exception Primary key ID missing from row or is null.
+	 */
+	public function save_many( $models ) {
```

Comment:
Leonidas and I discussed that we should go for a insert_many instead of a save_many because  we wouldn't be able to track down which query failed. Also adding a transaction would be detrimental for performance. So we will do inserts only and assume that all models are new. We should throw an exception if any of the models are not new.

Reply:
Renamed it into insert_many and now it throws an exception when non-new instances are passed.